lib/benchmark/benchmark.go

```diff
@@ -0,0 +1,112 @@
+package benchmark
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/HdrHistogram/hdrhistogram-go"
+	"github.com/gravitational/teleport/lib/client"
+	logrus "github.com/sirupsen/logrus"
+)
+
+const (
+	// PauseTimeBetweenBenchmarks is the time to pause between each benchmark
+	PauseTimeBetweenBenchmarks = time.Second * 5
+)
+
+// Generator provides a standardized way to get successive benchmarks from a consistent interface
+type Generator interface {
+	Generator() bool
+	GetBenchmark() (context.Context, Config, error)
+	Benchmark([]string, *client.TeleportClient) ([]*Result, error)
+}
+
+// Run is used to run the benchmarks, it is given a generator, command to run,
+// a host, host login, and proxy. If host login or proxy is an empty string, it will
+// use the default login
+func Run(cnf interface{}, cmd, host, login, proxy string) ([]*Result, error) {
```

Comment:
can we return `[]Result` and also, is there a better way than making `cnf interface{}`